parbase/FairParRootFileIo.cxx

```diff
@@ -147,33 +151,31 @@ Bool_t FairParRootFileIo::open(const Text_t* fname, Option_t* option, const Text
 
 Bool_t FairParRootFileIo::open(const TList* fnamelist, Option_t* option, const Text_t* ftitle, Int_t compress)
 {
-    TDatime currentDate;
-    TString newParFileName = "";
-    TFile* newParFile = 0;
-
+    char fname[100];
+    strncpy(fname, "/tmp/allParams_XXXXXX.root", 100);
+    int fd=mkstemps(fname, 5);
+    if (fd==-1)
+      throw std::runtime_error("mkstemps failed!");
+    ::close(fd);
+    TFile* newParFile = TFile::Open(fname, "RECREATE");
```

Comment:
Thinking about the race condition you point out, is there actually any need to have a filesystem handle for this `allParams...` file at all (also who is cleaning up such files?)? What about using a [`TMemFile`](https://root.cern/doc/master/classTMemFile.html) and avoid using the filesystem alltogether (I am not too familiar with the `FairParRootFileIo`)?

Reply:
It looks like `FairRunAnaProof::RunOnProof` actually uses the filename (via `::getFilename()`) for something. And from a quick looks, this *seems* to make its way to `FairAnaSelector::Init`, where it looks like it is used for actually opening the file (again).

Might all be bogus and not used by anybody / not happen in the real world. But please take this into account.

Reply:
Then I get the impression this generated *allParams* root file is not really something that should be dumped to `/tmp` tbh